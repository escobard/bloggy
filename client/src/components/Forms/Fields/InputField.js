import React from "react";
import PropTypes from "prop-types";
import TextField from "material-ui/TextField";

import { Field } from "redux-form";

const InputField = ({hint, label, name}) => {
	return (
		<div className="input-field">
			<TextField
				hintText={hint}
				floatingLabelText={label}
			>
				<Field type="text" name={name} component="input" />
			</TextField>
		</div>
	);
};

InputField.defaultProps = {
	hint: "test hint",
	label: "test label",
	name: "test name"
};

InputField.propTypes = {
	hint: PropTypes.string,
	label: PropTypes.string
};

export default InputField;