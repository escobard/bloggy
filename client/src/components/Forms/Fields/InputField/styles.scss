@import "../../../../styles/variables";
@import "../../../../styles/mixins";

.input-field{
	.input-container{
		width: 100% !important;
	}
}