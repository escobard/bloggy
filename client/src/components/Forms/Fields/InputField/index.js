import React from "react";
import PropTypes from "prop-types";
import TextField from "material-ui/TextField";
import { Field } from "redux-form";

import styles from "./styles.scss"

const InputField = ({hint, label, name}) => {
	return (
		<div className="input-field">
			<TextField
				hintText={hint}
				floatingLabelText={label}
				className="input-container"
			>
				<Field type="text" name={name} component="input" />
			</TextField>
		</div>
	);
};

InputField.defaultProps = {
	hint: "test hint",
	label: "test label",
	name: "test name"
};

InputField.propTypes = {
	hint: PropTypes.string,
	label: PropTypes.string
};

export default InputField;