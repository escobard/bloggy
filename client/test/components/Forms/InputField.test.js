import React, { Component } from 'react'
import InputField from '../../src/components/Forms/Fields/InputField';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'

describe('InputField snapshot', () => {
    const initialState = {output:100}
    const mockStore = configureStore()
    let store,container

    beforeEach(()=>{
        store = mockStore(initialState)
        container = shallow(<InputField /> )  
    })

    it('+++ render the InputField component', () => {
       expect(toJson(container)).toMatchSnapshot()
    });

});