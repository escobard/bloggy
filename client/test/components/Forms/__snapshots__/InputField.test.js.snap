// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField snapshot +++ render the InputField component 1`] = `
<div
  className="input-field"
>
  <TextField
    className="input-container"
    disabled={false}
    floatingLabelFixed={false}
    floatingLabelText="test label"
    fullWidth={false}
    hintText="test hint"
    multiLine={false}
    rows={1}
    type="text"
    underlineShow={true}
  >
    <Field
      component="input"
      name="test name"
      type="text"
    />
  </TextField>
</div>
`;
